Mathlib/AlgebraicTopology/SimplicialSet/StrictSegal.lean

```diff
@@ -386,6 +427,78 @@ lemma spine_δ_arrow_eq (h : j = i.succ.castSucc) :
   rw [mkOfSucc_δ_eq h, spineToSimplex_edge]
 
 end StrictSegal
+
+/-- Helper structure in order to show that a simplicial set is strict Segal. -/
+structure StrictSegalCore (n : ℕ) where
+  /-- Map which produces a `n + 1`-simplex from a `1`-simplex and a `n`-simplex when
+  the target vertex of the `1`-simplex equals the zeroth simplex of the `n`-simplex. -/
+  concat (x : X _⦋1⦌) (s : X _⦋n⦌) (h : X.δ 0 x = X.map ((SimplexCategory.const _ _ 0).op) s) :
+    X _⦋n + 1⦌
+  map_mkOfSucc_zero_concat x s h : X.map ((mkOfSucc 0).op) (concat x s h) = x
+  δ₀_concat x s h : X.δ 0 (concat x s h) = s
+  injective {x y : X _⦋n + 1⦌} (h : X.map ((mkOfSucc 0).op) x = X.map ((mkOfSucc 0).op) y)
+    (h₀ : X.δ 0 x = X.δ 0 y) : x = y
+
+namespace StrictSegalCore
+
+variable {X} (h : ∀ n, X.StrictSegalCore n) {n : ℕ} (p : X.Path n)
+
+/-- Auxiliary definition for `StrictSegalCore.spineToSimplex`. -/
+def spineToSimplexAux : { s : X _⦋n⦌ // X.spine _ s = p } := by
+  induction n with
+  | zero => exact ⟨p.vertex 0, by aesop⟩
+  | succ n hn =>
+    refine ⟨(h n).concat (p.arrow 0) (hn (p.interval 1 n)).1 ?_, ?_⟩
+    · rw [p.arrow_tgt 0]
+      exact Path.congr_vertex ((hn (p.interval 1 n)).2).symm 0
+    · ext i
+      obtain rfl | ⟨i, rfl⟩ := i.eq_zero_or_eq_succ
+      · dsimp
+        rw [map_mkOfSucc_zero_concat]
+      · simp only [spine_arrow, ← SimplexCategory.mkOfSucc_δ_gt (j := 0) (i := i) (by simp),
+          op_comp, FunctorToTypes.map_comp_apply, ← SimplicialObject.δ_def, δ₀_concat,
+          ← p.arrow_interval 1 n i i.succ (by omega) (by simp only [Fin.val_succ]; omega)]
+        exact Path.congr_arrow (hn (p.interval 1 n)).2 i
+
+/-- Auxiliary definition for `StrictSegal.ofCore`. -/
+def spineToSimplex : X _⦋n⦌ := (spineToSimplexAux h p).1
+
+@[simp]
+lemma spine_spineToSimplex : X.spine n (spineToSimplex h p) = p := (spineToSimplexAux h p).2
+
+lemma spineToSimplex_zero (p : X.Path 0) : spineToSimplex h p = p.vertex 0 := rfl
+
+lemma spineToSimplex_succ (p : X.Path (n + 1)) :
+    spineToSimplex h p = (h n).concat (p.arrow 0) (spineToSimplex h (p.interval 1 n)) (by
+      rw [p.arrow_tgt 0]
+      exact Path.congr_vertex (spine_spineToSimplex h (p.interval 1 n)).symm 0) := rfl
```

Comment:
```suggestion
      exact Path.congr_vertex (spine_spineToSimplex h (p.interval 1 n)).symm 0) := 
  rfl
```
I think it’s a case where having the `rfl` proof on a new line is more readable.